src/dbnode/storage/index/block.go

```diff
@@ -590,7 +592,16 @@ func (b *block) docWithinQueryRange(doc doc.Document, opts QueryOptions) bool {
 	} else {
 		b.metrics.entryUnreconciledOnQuery.Inc(1)
 	}
+
 	defer closer.Close()
+	if reconciled && onIndexSeries != md.OnIndexSeries {
+		// NB: attempt to reconcile the index series here, especially in the case
+		// of foreground segments, since reconciling duplicate index entry writes
+		// only happens when the entry has been moved to background segment
+		// processing, and compacted.
+		b.metrics.entryReconcilingDuplicates.Inc(1)
+		md.OnIndexSeries.TryReconcileDuplicates()
```

Comment:
Maybe all of this logic can be included in this `ReconciledOnIndexSeries` func above? Seems more clear since the caller is expecting that this entry is fully "correct" that gets returned

Reply:
+1

Reply:
:+1: sounds good